main/com.spotify/eval/scenarios.txt

```diff
@@ -30,3 +30,73 @@ U: now => (@com.spotify.song(), id =~ "destination calabria" && contains~(artist
 
 A: Alright, I played Destination Calabria( - Radio Edit)?( on Spotify)?( on (Coolest Computer|giovanni-xps15))? ?.
 A: >> expecting = null
+
+====
+# 4-from-album (req=com.spotify)
+
+# play songs from the album abbey road
+U: \t $dialogue @org.thingpedia.dialogue.transaction.execute;
+U: now => (@com.spotify.song(), album =~ "abbey road") => @com.spotify.play_song(song=id);
+
+A: Alright, your request was completed successfully\.?
+A: >> expecting = null
+
+====
+# 5-from-album-artist (req=com.spotify)
+
+# play songs about jane by maroon 5
+U: \t $dialogue @org.thingpedia.dialogue.transaction.execute;
+U: now => (@com.spotify.song(), album =~ "songs about jane" && contains~(artists, "maroon 5")) => @com.spotify.play_song(song=id);
+
+A: Alright, your request was completed successfully\.?
+A: >> expecting = null
+
+====
+# 6-sort-popularity-artist (req=com.spotify)
+
+# play the most popular songs by post malone
+U: \t $dialogue @org.thingpedia.dialogue.transaction.execute;
+U: now => sort popularity desc of (@com.spotify.song(), contains~(artists, "post malone")) => @com.spotify.play_song(song=id);
```

Comment:
"most popular" is `(sort ...)[1]` (note the `[1]` at the end. See also https://github.com/stanford-oval/thingpedia-common-devices/pull/156/commits/6d0ad7dfaf67c4b68c4a6e1f711fe991fa1c0a98